catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -266,21 +261,74 @@ public void setTerrainProvider(String terrainProvider) {
     }
 
     private void setProxiesForImagery(List<String> imageryProviders) {
+        List<Map<String, Object>> newImageryProviders = new ArrayList<>();
+        for (String provider : imageryProviders) {
+            Map<String, Object> proxiedProvider = getProviderConfigMap(provider);
+            if (proxiedProvider != null) {
+                newImageryProviders.add(proxiedProvider);
+            }
+        }
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(proxiedImageryProviders, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, proxiedImageryProviders, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
+                .map(provider -> urlToProxyMap.get(provider))
+                .collect(Collectors.toList());
+
+        stopImageryEndpoints(proxiesToStop);
+        startImageryEndpoints(imageryProvidersToStart);
+        proxiedImageryProviders = newImageryProviders;
+    }
+
+    private void findDifferences(List<Map<String, Object>> innerList,
+            List<Map<String, Object>> outerList,
+            List<Map<String, Object>> differences) {
+        for (Map<String, Object> newProvider : outerList) {
+            boolean found = false;
+            for (Map<String, Object> provider : innerList) {
+                if (provider.get(URL).equals(newProvider.get(URL))) {
+                    found = true;
+                }
+            }
+            if (!found) {
+                differences.add(newProvider);
+            }
+        }
+    }
+
+    private void stopImageryEndpoints(List<String> imageryEndpoints) {
         if (imageryEndpoints.size() > 0) {
             for (String endpoint : imageryEndpoints) {
                 try {
                     httpProxy.stop(endpoint);
+                    urlToProxyMap.remove(endpoint);
+                    imageryEndpoints.remove(endpoint);
                 } catch (Exception e) {
                     LOGGER.error("Unable to stop proxy endpoint.", e);
```

Comment:
Is there any value in including the name of the endpoint that failed on this iteration through?